baseplate/secrets/fetcher.py

```diff
@@ -108,16 +118,72 @@ def ttl_to_time(ttl):
 
 class VaultClientFactory(object):
     """Factory that makes authenticated clients."""
-    def __init__(self, base_url, role):
+    def __init__(self, base_url, role, auth_type, mount_point):
         self.base_url = base_url
         self.role = role
+        self.auth_type = auth_type
+        self.mount_point = mount_point
         self.session = requests.Session()
         self.client = None
 
     def _make_client(self):
-        """Authenticate with Vault and return a client containing that token.
+        """Obtain a client token from an auth backend and return a Vault client with it."""
+        client_token, lease_duration = self.auth_type(self)
+
+        return VaultClient(
+            self.session,
+            self.base_url,
+            client_token,
+            lease_duration,
+        )
+
+    def _vault_kubernetes_auth(self):
+        """Gets a client token from Vault through the Kubernetes auth backend.
```

Comment:
sorry to be so picky, but for consistency pep-257 says it should be in the imperative i.e. "Get" vs "Gets".

Reply:
no problem, consistency is good